Fix #19906: Destructible cheat does not allow ride modification

Chase-Percy opened the pull request:
- Fixes the all-destructible cheat issue GH-19906.
- Adds a variable to represent if the indestructible flags should be ignored.
- Remove the old function that altered the flags on all the rides.
- Makes the cheat toggleable, but I'm not sure if this is desired.
- Added serialization and deserialization for the new cheat variable.

tupaschoal commented on src/openrct2/actions/TrackRemoveAction.cpp:
```diff
@@ -114,7 +114,7 @@ GameActions::Result TrackRemoveAction::Query() const
     }
 
-    if (tileElement->AsTrack()->IsIndestructible())
+    if (tileElement->AsTrack()->IsIndestructible() && !gCheatsMakeAllDestructible)
```
Maybe the `isIndestructible` function itself should check the cheat?

Chase-Percy replied:
Yup makes sense, [80eb038](https://github.com/OpenRCT2/OpenRCT2/pull/19924/commits/80eb038c24e20cb9dc01e4e2a52b4374b4c2a429)

tupaschoal reviewed (commented):
> Makes the cheat toggleable, but I'm not sure if this is desired.

I'd say yes, but then maybe it also needs to be come a checkbox on the cheats window, instead of a button, since no button is togglable there.

Chase-Percy commented:
> > Makes the cheat toggleable, but I'm not sure if this is desired.
> 
> I'd say yes, but then maybe it also needs to be come a checkbox on the cheats window, instead of a button, since no button is togglable there.

Nice, yea makes it a lot cleaner: [8d83676](https://github.com/OpenRCT2/OpenRCT2/pull/19924/commits/8d8367686c8ee0b3edd6a3b5c52b4861af94810c)

tupaschoal reviewed (commented):
I've tested it and it works great, with the added benefit of being able to restore no "non-destructible" and a bonus of having the ride tab of cheats with symmetrical buttons at the top now :D

![image](https://user-images.githubusercontent.com/6443427/232029742-a9036bc9-c2c7-49c7-8fbe-bf1fe55c40ed.png)

Can you please add a changelog entry under https://github.com/OpenRCT2/OpenRCT2/blob/60ac375e31d86dbd5b5ca7d10803dfd84cfb3fd5/distribution/changelog.txt ?

@OpenRCT2/developers do you think we should bump the network version on? I mean, people can have different cheat behavior on multiplayer, but it's not like it's game breaking, so I'm inclined not to.

PS: If we decide to bump the network verison it will be her: https://github.com/OpenRCT2/OpenRCT2/blob/60ac375e31d86dbd5b5ca7d10803dfd84cfb3fd5/src/openrct2/network/NetworkBase.cpp#L46

ZehMatt commented:
The PR looks fine but it does need to bump up the park version and network version, the cheat is saved inside the park file so that needs a save version bump, and the logic was changed for game actions so we need a network bump. Thanks for the PR also.

Chase-Percy commented:
- Added the changelog info: [426b499](https://github.com/OpenRCT2/OpenRCT2/pull/19924/commits/426b499235414e443cc6cb2daf1ee5db3e1773dd)
- Added my name to contributors under fixes: [dc9aa67](https://github.com/OpenRCT2/OpenRCT2/pull/19924/commits/dc9aa67d3a5cd17dba9e0768cb77d3153f3f80ae)
- Increased the network version (10 -> 11): [8502d32](https://github.com/OpenRCT2/OpenRCT2/pull/19924/commits/8502d3207870ba2d1877d563553f88105a589d58)
- Increased the park save versions (27 -> 28): [f5c0db5](https://github.com/OpenRCT2/OpenRCT2/pull/19924/commits/f5c0db5846c1b812c8e3757250cbbf6b21c883e9). I also increased the min compatible version since old saves that used the cheat altered the rides flags which had no option to undo it.

Let me know if there is anything else I need to tidy up, thanks!

tupaschoal commented:
Thank you!
